src/other/index.jl

```diff
@@ -225,11 +226,29 @@ struct SubIndex{I<:AbstractIndex,S<:AbstractVector{Int},T<:AbstractVector{Int}}
     remap::T # reverse of cols
 end
 
-parentcols(ind::SubIndex) = ind.cols
-
 SubIndex(parent::AbstractIndex, ::Colon) = parent
 
-@inline function SubIndex(parent::AbstractIndex, cols::AbstractUnitRange{Int})
+@inline parentcols(ind::SubIndex) = ind.cols
+
+Base.@propagate_inbounds parentcols(ind::SubIndex, idx::Union{Integer, AbstractVector{<:Integer}}) =
```

Comment:
Would it makes sense to return a view for `idx::AbstractVector`?

Reply:
Good point. It should not break anything and will be somewhat faster.